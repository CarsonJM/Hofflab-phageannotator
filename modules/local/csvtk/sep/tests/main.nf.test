// nf-core modules test csvtk/concat
nextflow_process {

    name "Test Process CSVTK_SEP"
    script "../main.nf"
    process "CSVTK_SEP"
    config "./nextflow.config"

    test("tsv - sep - tsv") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [ file("${projectDir}/assets/test-datasets/csvtk/sep_test.tsv", checkIfExists: true) ]
                ]
                input[1] = "tsv"
                input[2] = "tsv"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("tsv - concat - tsv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [ file("${projectDir}/assets/test-datasets/csvtk/sep_test.tsv", checkIfExists: true) ]
                ]
                input[1] = "tsv"
                input[2] = "tsv"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
