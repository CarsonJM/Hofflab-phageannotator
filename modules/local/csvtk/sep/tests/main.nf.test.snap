{
    "tsv - sep - tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_sep.tsv:md5,83716e72d7b48a4415af0b91cb5442a9"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b980d848912e7b6792adf66e6f4396b3"
                ],
                "csv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_sep.tsv:md5,83716e72d7b48a4415af0b91cb5442a9"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b980d848912e7b6792adf66e6f4396b3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-06-19T11:23:30.141659909"
    },
    "tsv - concat - tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_sep.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b980d848912e7b6792adf66e6f4396b3"
                ],
                "csv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_sep.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b980d848912e7b6792adf66e6f4396b3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-06-19T11:23:36.440738307"
    }
}